table/iterator.cc

```diff
@@ -19,10 +19,18 @@ Cleanable::Cleanable() {
   cleanup_.next = nullptr;
 }
 
-Cleanable::~Cleanable() {
+Cleanable::~Cleanable() { DoCleanup(); }
+
+void Cleanable::Reset() {
```

Comment:
I don't think the 2 extra assignments should be a concern (except if benchmarks showed that they are of course)
This was just a suggestion, final decision is up to you